src/node/nodestate.h

```diff
@@ -1070,13 +1034,19 @@ namespace ccf
       return true;
     }
 
-    // Used from nodefrontend.h to set the joiner's fresh key to encrypt past
-    // network secrets
-    void set_joiner_key(
-      NodeId joiner_id, const std::vector<uint8_t>& raw_key) override
+    bool rekey_ledger(Store::Tx& tx) override
     {
-      LOG_DEBUG_FMT("Setting fresh key for joiner {}", joiner_id);
-      joiners_fresh_keys.emplace(joiner_id, raw_key);
+      std::lock_guard<SpinLock> guard(lock);
+      sm.expect(State::partOfNetwork);
+
+      // Effects of ledger rekey are only observed for the next transaction,
```

Comment:
Should this be "observed **from** the next transaction"?

Reply:
Absolutely!